pkg/services/dashboards/database/database_dashboard_public_test.go

```diff
@@ -0,0 +1,157 @@
+				},
+			},
+		})
+		require.NoError(t, err)
+
+		pdc, err := dashboardStore.GetPublicDashboardConfig(savedDashboard.OrgId, savedDashboard.Uid)
+		require.NoError(t, err)
+		assert.Equal(t, resp, pdc)
+	})
+}
+
+func TestSavePublicDashboard(t *testing.T) {
+	var sqlStore *sqlstore.SQLStore
+	var dashboardStore *DashboardStore
+	var savedDashboard *models.Dashboard
+
+	setup := func() {
+		sqlStore = sqlstore.InitTestDB(t, sqlstore.InitTestDBOpt{FeatureFlags: []string{featuremgmt.FlagPublicDashboards}})
+		dashboardStore = ProvideDashboardStore(sqlStore)
+		savedDashboard = insertTestDashboard(t, dashboardStore, "testDashie", 1, 0, true)
+	}
+
+	t.Run("saves new public dashboard", func(t *testing.T) {
+		setup()
+		resp, err := dashboardStore.SavePublicDashboardConfig(models.SavePublicDashboardConfigCommand{
+			DashboardUid: savedDashboard.Uid,
+			OrgId:        savedDashboard.OrgId,
+			PublicDashboardConfig: models.PublicDashboardConfig{
+				IsPublic: true,
+				PublicDashboard: models.PublicDashboard{
+					DashboardUid: savedDashboard.Uid,
+					OrgId:        savedDashboard.OrgId,
+				},
+			},
+		})
+		require.NoError(t, err)
+
+		pdc, err := dashboardStore.GetPublicDashboardConfig(savedDashboard.OrgId, savedDashboard.Uid)
+		require.NoError(t, err)
+
+		//verify saved response and queried response are the same
+		assert.Equal(t, resp, pdc)
+
+		// verify we have a valid uid
+		assert.True(t, util.IsValidShortUID(pdc.PublicDashboard.Uid))
+	})
+
+	t.Run("returns ErrDashboardIdentifierNotSet", func(t *testing.T) {
+		setup()
+		_, err := dashboardStore.SavePublicDashboardConfig(models.SavePublicDashboardConfigCommand{
+			DashboardUid: savedDashboard.Uid,
+			OrgId:        savedDashboard.OrgId,
+			PublicDashboardConfig: models.PublicDashboardConfig{
+				IsPublic: true,
+				PublicDashboard: models.PublicDashboard{
+					DashboardUid: "",
+					OrgId:        savedDashboard.OrgId,
+				},
+			},
+		})
+		require.Error(t, models.ErrDashboardIdentifierNotSet, err)
+	})
+
+	t.Run("overwrites existing public dashboard", func(t *testing.T) {
+		setup()
+
+		pdUid := util.GenerateShortUID()
+
+		// insert initial record
+		_, err := dashboardStore.SavePublicDashboardConfig(models.SavePublicDashboardConfigCommand{
+			DashboardUid: savedDashboard.Uid,
+			OrgId:        savedDashboard.OrgId,
+			PublicDashboardConfig: models.PublicDashboardConfig{
+				IsPublic: true,
+				PublicDashboard: models.PublicDashboard{
+					Uid:          pdUid,
+					DashboardUid: savedDashboard.Uid,
+					OrgId:        savedDashboard.OrgId,
+				},
+			},
+		})
+		require.NoError(t, err)
+
+		// update initial record
+		resp, err := dashboardStore.SavePublicDashboardConfig(models.SavePublicDashboardConfigCommand{
+			DashboardUid: savedDashboard.Uid,
+			OrgId:        savedDashboard.OrgId,
+			PublicDashboardConfig: models.PublicDashboardConfig{
+				IsPublic: false,
+				PublicDashboard: models.PublicDashboard{
+					Uid:           pdUid,
```

Comment:
this uid that is embedded inside the json bit, what is it used for?

Reply:
If there is an existing public dashboard with that uid, we update it.